packages/web3-providers/src/providers/HttpProvider.js

```diff
@@ -180,7 +180,11 @@ export default class HttpProvider {
 
                 reject(new Error(`Connection error: Timeout exceeded after ${this.timeout}ms`));
             };
-
+            
+            request.onerror = () => {
+                this.connected = false;
+                reject(new Error(`Network error`))
+            }
```

Comment:
could you add the missing break here and the missing semicolon on line 186.

Reply:
@nivida done